python/jupyterlab-chat/jupyterlab_chat/models.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from dataclasses import dataclass
+from typing import Literal, Optional
+from jupyter_server.auth import User as JupyterUser
+
+
+@dataclass(kw_only=True)
+class Message:
+    """ Object representing a message """
+
+    type: Literal["msg"] = "msg"
+    body: str
+    """ The content of the message """
+
+    id: str
+    """ Unique ID """
+
+    time: float
+    """ Timestamp in second since epoch """
+
+    sender: str
+    """ The message sender unique id """
+
+    raw_time: Optional[bool] = None
+    """ Whether the timestamp is raw (from client) or not (from server, unified) """
+
+    deleted: Optional[bool] = None
+    """ Whether the message has been deleted or not (body should be empty if True) """
+
+    edited: Optional[bool] = None
+    """ Whether the message has been edited or not """
```

Comment:
Do the boolean fields here need to be `Optional`? It seems to make more sense to ensure `raw_time` is set at init, and default to `False` for `deleted` and `edited`. This would allow the PR to be simplified by removing a few `getattr()` calls.

Reply:
The idea of using `Optional` was to avoid saving extra information to the file.
I thought it was automatic but not, I added this in https://github.com/jupyterlab/jupyter-chat/pull/119/commits/07b692681454271d09d5e55f811bf03e6f396b86

Reply:
Sounds reasonable. Can we document the default value for each of these fields in the docstrings?

Reply:
BTW you can use `bool | None` instead of `Optional[bool]`, provided that you add this line at the top of the file:
```py
from __future__ import annotations
```